### PR TITLE
dotCMS/core#23457 Containers "clear" button clear all the fields

### PR DESCRIPTION
**Pull request by zulqarnainvd**
### Proposed Changes
* clear button clear all fields except name, description, and max-content
* save button disabled if data is not changed in edit container.

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.spec.ts [hunk 1]
```diff
@@ -204,13 +246,18 @@ describe('DotContainerPropertiesComponent', () => {
 
         it('should focus on title field', async () => {
             const inplace = de.query(By.css('[data-testId="inplace"]'));
+            inplace.componentInstance.activate();
+            fixture.detectChanges();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.spec.ts:250]: Why? looks like you are manually activating the focus here and it should be auto.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts [hunk 2]
```diff
@@ -60,9 +60,13 @@ export class DotContainerPropertiesComponent implements OnInit {
                 });
 
                 this.addContainerFormControl(containerStructures);
+                if (this.form.value?.identifier)
+                    this.store.updateOriginalFormState(this.form.value);
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:64]: We try to follow a more concise code styling, so please use `{}`

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts [hunk 3]
```diff
@@ -60,9 +60,13 @@ export class DotContainerPropertiesComponent implements OnInit {
...
             });
+
         this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((values) => {
             this.store.updateIsContentTypeButtonEnabled(values.maxContentlets > 0);
+            this.store.updateFormStatus({ invalidForm: !this.form.valid, container: values });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:69]: You can do this in one state update.

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:69]: Keeps the rerenders at min as possible.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts [hunk 4]
```diff
@@ -180,14 +187,19 @@ export class DotContainerPropertiesComponent implements OnInit {
...
-                this.store.updateContentTypeVisibility(false);
-                this.form.reset();
                 this.form.get('containerStructures').clearValidators();
                 this.form.get('containerStructures').reset();
+                this.form.get('preLoop').reset();
+                this.form.get('postLoop').reset();
                 // clear containerStructures array
                 (this.form.get('containerStructures') as FormArray).clear();
                 this.form.get('code').addValidators(Validators.required);
                 this.form.updateValueAndValidity();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:197]: Can you move this to a private method and add a name that can be more explicit in what we do.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.spec.ts (hunk 1 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.spec.ts:250]: it is not activating the focus it shows the field of the title. In edit mode, you only see the title text, not the title field.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts (hunk 2 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:64]: Noted

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts (hunk 4 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts:197]: Noted

Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts]: updated

File: core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts (hunk 3 above)
Comment (zulqarnainvd) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts]: updated

**Review by jdotcms — APPROVED** [review]

**Pull request merged**